CMakeLists.txt

```diff
@@ -0,0 +1,613 @@
+
+# Write ares_config.h configuration file.  This is used only for the build.
+CONFIGURE_FILE (ares_config.h.cmake ${PROJECT_BINARY_DIR}/ares_config.h)
+
+
+INCLUDE_DIRECTORIES (
+	${CARES_INCLUDE_DIRS}
+)
+
+# Source files.
+SET (CARES_SOURCES
+	ares__close_sockets.c
+	ares__get_hostent.c
+	ares__read_line.c
+	ares__timeval.c
+	ares_cancel.c
+	ares_data.c
+	ares_destroy.c
+	ares_expand_name.c
+	ares_expand_string.c
+	ares_fds.c
+	ares_free_hostent.c
+	ares_free_string.c
+	ares_getenv.c
+	ares_gethostbyaddr.c
+	ares_gethostbyname.c
+	ares_getnameinfo.c
+	ares_getsock.c
+	ares_init.c
+	ares_library_init.c
+	ares_llist.c
+	ares_mkquery.c
+	ares_create_query.c
+	ares_nowarn.c
+	ares_options.c
+	ares_parse_a_reply.c
+	ares_parse_aaaa_reply.c
+	ares_parse_mx_reply.c
+	ares_parse_naptr_reply.c
+	ares_parse_ns_reply.c
+	ares_parse_ptr_reply.c
+	ares_parse_soa_reply.c
+	ares_parse_srv_reply.c
+	ares_parse_txt_reply.c
+	ares_platform.c
+	ares_process.c
+	ares_query.c
+	ares_search.c
+	ares_send.c
+	ares_strcasecmp.c
+	ares_strdup.c
+	ares_strerror.c
+	ares_timeout.c
+	ares_version.c
+	ares_writev.c
+	bitncmp.c
+	inet_net_pton.c
+	inet_ntop.c
+	windows_port.c
+)
+
+# Build the dynamic/shared library
+IF (${CARES_SHARED})
+	ADD_LIBRARY (${PROJECT_NAME} SHARED ${CARES_SOURCES})
+	SET_TARGET_PROPERTIES (${PROJECT_NAME} PROPERTIES COMPILE_FLAGS "-DHAVE_CONFIG_H=1 -DCARES_BUILDING_LIBRARY")
+
+	# Convert CARES_LIB_VERSIONINFO libtool version format into VERSION and SOVERSION
+	# Convert from ":" separated into CMake list format using ";"
+	STRING (REPLACE ":" ";" CARES_LIB_VERSIONINFO ${CARES_LIB_VERSIONINFO})
+	LIST (GET CARES_LIB_VERSIONINFO 0 CARES_LIB_VERSION_CURRENT)
+	LIST (GET CARES_LIB_VERSIONINFO 1 CARES_LIB_VERSION_REVISION)
+	LIST (GET CARES_LIB_VERSIONINFO 2 CARES_LIB_VERSION_AGE)
+	MATH (EXPR CARES_LIB_VERSION_MAJOR "${CARES_LIB_VERSION_CURRENT} - ${CARES_LIB_VERSION_AGE}")
+	SET  (CARES_LIB_VERSION_MINOR "${CARES_LIB_VERSION_AGE}")
+	SET  (CARES_LIB_VERSION_RELEASE "${CARES_LIB_VERSION_REVISION}")
+	SET_TARGET_PROPERTIES (${PROJECT_NAME} PROPERTIES VERSION "${CARES_LIB_VERSION_MAJOR}.${CARES_LIB_VERSION_MINOR}.${CARES_LIB_VERSION_RELEASE}")
+	SET_TARGET_PROPERTIES (${PROJECT_NAME} PROPERTIES SOVERSION "${CARES_LIB_VERSION_MAJOR}")
+
+	TARGET_LINK_LIBRARIES (${PROJECT_NAME} ${CARES_DEPENDENT_LIBS})
+	IF (${CARES_INSTALL})
+		INSTALL (TARGETS ${PROJECT_NAME} DESTINATION ${CARES_INSTALL_LOCATION_LIBS})
+	ENDIF ()
+
+	# Set a static suffix if building both shared and static libs
+	SET (STATIC_SUFFIX "_static")
+ENDIF ()
+
+# Build the static library
+IF (${CARES_STATIC})
+	ADD_LIBRARY (${PROJECT_NAME}${STATIC_SUFFIX} STATIC ${CARES_SOURCES})
+	SET_TARGET_PROPERTIES (${PROJECT_NAME}${STATIC_SUFFIX} PROPERTIES COMPILE_FLAGS "-DHAVE_CONFIG_H=1 -DCARES_STATICLIB" OUTPUT_NAME ${PROJECT_NAME})
+	IF (${CARES_STATIC_PIC})
+		SET_TARGET_PROPERTIES (${PROJECT_NAME}${STATIC_SUFFIX} PROPERTIES POSITION_INDEPENDENT_CODE True)
```

Comment:
Instead of setting it per-target you could set ``CMAKE_POSITION_INDEPENDENT_CODE`` before any creating the first target. It will populate the per-target value.